edge-modules/edge-dashboard/src/lib.rs

```diff
@@ -229,6 +234,75 @@ fn restart_module(
+    let api_ver = &info.api_version;
+
+    let response = context
+        .edge_config
+        .as_ref()
+        .map(move |config| {
+            let mgmt_uri = config.connect().management_uri();
+            Either::A(
+                Url::parse(&format!("{}/modules/?api-version={}", mgmt_uri, api_ver))
+                    .map_err(ErrorInternalServerError)
+                    .and_then(|url| ModuleClient::new(&url).map_err(ErrorInternalServerError)) // can't connect to the endpoint
+                    .map(move |mod_client| {
+                        mod_client
+                            .logs(module_id, &LogOptions::new())
+                            .map_err(ErrorInternalServerError)
+                            .and_then(|data| {
+                                data.map_err(ErrorInternalServerError)
+                                    .fold(BytesMut::new(), |mut acc, chunk| {
+                                        acc.extend_from_slice(chunk.as_ref());
+                                        Ok::<_, ActixError>(acc)
+                                    })
+                                    .and_then(|body| {
+                                        let mut clone = body.freeze().to_vec().clone();
+                                        let mut curr = true;
+                                        let mut next = true;
+                                        // if the next byte after the \0 encodes the length of msg, ignore
+                                        clone.retain(|&byte| {
+                                            curr = next;
+                                            next = byte as char != '\0';
+                                            byte != 0 && byte != 1 && curr && byte as char != '+'
+                                        });
+                                        clone.retain(|&byte| (byte as char).is_ascii());
+                                        if let Ok(content) = String::from_utf8(clone) {
+                                            HttpResponse::Ok().body(content)
+                                        } else {
+                                            HttpResponse::ServiceUnavailable()
+                                                .body("Logs unable to be displayed")
+                                        }
+                                    })
+                            })
+                    })
+                    .into_future()
+                    .flatten(),
+            )
+        })
+        .unwrap_or_else(|err| {
+            Either::B(ok(HttpResponse::ServiceUnavailable()
+                .content_type("text/plain")
+                .body(format!("{:?}", err))))
+        });
+
+    Box::new(response)
+}
+
+fn get_diagnostics() -> HttpResponse {
+    let output = Command::new("iotedge")
+        .args(&["check", "--output", "json"])
+        .output()
+        .expect("Failed to execute command");
```

Comment:
Rather than calling expect, perhaps generate a repsonse:
```Rust
fn get_diagnostics() -> HttpResponse {
    Command::new("iotedge")
        .args(&["check", "--output", "json"])
        .output()
        .map( |o|  HttpResponse::Ok().body(o.stdout))
        .unwrap_or_else( |_| HttpResponse::ServiceUnavailable().body("Failed to execute command"))
}
```